mergin/test/test_client.py

```diff
@@ -805,3 +806,136 @@ def test_logging(mc):
+    project_dir = os.path.join(TMP_DIR, test_project)
+    f_updated = "base.gpkg"
+
+    mp = create_versioned_project(mc, test_project, project_dir, f_updated, remove=False)
+
+    project_info = mc.project_info(project)
+    assert project_info["version"] == "v4"
+    file_history = mc.project_file_history_info(project, f_updated)
+
+    wrong_versions_err = "Wrong version parameter: 1-5 while getting diffs for base.gpkg. Available versions: [1, 2, 3, 4]"
+    with pytest.raises(ClientError) as err:
+        mod_versions = get_versions_with_file_changes(
+            mc, project, f_updated, version_from="v1", version_to="v5", file_history=file_history
+        )
+    assert err.value.args[0] == wrong_versions_err
+
+    mod_versions = get_versions_with_file_changes(
+        mc, project, f_updated, version_from="v2", version_to="v4", file_history=file_history
+    )
+    assert mod_versions == [f"v{i}" for i in range(2, 5)]
+
+
+def test_download_file(mc):
+    """Test downloading single file at specified versions."""
+    test_project = 'test_download_file'
+    project = API_USER + '/' + test_project
+    project_dir = os.path.join(TMP_DIR, test_project)
+    f_updated = "base.gpkg"
+
+    mp = create_versioned_project(mc, test_project, project_dir, f_updated)
+
+    project_info = mc.project_info(project)
+    assert project_info["version"] == "v5"
+
+    # Download the base file at versions 2-4 and check the changes
+    expected = {
+        2: {"table": "simple", "change": "insert", "nr_of_changes": 1},
+        3: {"table": "simple", "change": "update", "nr_of_changes": 2},
+        4: {"table": "simple", "change": "update", "nr_of_changes": 3},
+    }
+    f_versioned = os.path.join(project_dir, f_updated)
+    for ver in range(2, 5):
+        if not os.path.exists(f_versioned):
+            shutil.copy(os.path.join(TEST_DATA_DIR, f_updated), mp.fpath_meta(f_updated))
+        else:
+            shutil.copy(f_versioned, mp.fpath_meta(f_updated))
+        mc.download_file(project_dir, f_updated, f_versioned, version=f"v{ver}")
+        diff_file = mp.fpath_meta(f"changes_v{ver}.diff")
+        mp.geodiff.create_changeset(
+            mp.fpath_meta(f_updated), f_versioned, mp.fpath_meta(diff_file))
+
+        assert mp.geodiff.has_changes(diff_file)
+        assert mp.geodiff.changes_count(diff_file) == expected[ver]["nr_of_changes"]
+
+        changes_file = diff_file + ".changes"
+        mp.geodiff.list_changes_summary(diff_file, changes_file)
+        with open(changes_file, 'r') as f:
+            expected_changed_table = expected[ver]["table"]
+            expected_change = expected[ver]["change"]
+            expected_nr_of_changes = expected[ver]["nr_of_changes"]
+            changes = json.loads(f.read())["geodiff_summary"][0]
+            assert changes["table"] == expected_changed_table
+            assert expected_change in changes
+            assert changes[expected_change] == expected_nr_of_changes
```

Comment:
This test is relatively complicated - 1. it has for cycle, 2. it has if-else branching, 3. it does quite complicated testing of what we expected vs what we got.

Can we please simplify it: 1. use one function to check whether two gpkgs have equal content (checks that create_changeset returns empty diff), 2. add to git expected gpkg for v2/3/4 and use asserts to check expected vs actual.